app/provider/app.go

```diff
@@ -352,9 +353,7 @@ func New(
 	// Remove the ConsumerRewardsPool from the group of blocked recipient addresses in bank
 	// this is required for the provider chain to be able to receive tokens from
 	// the consumer chain
-	bankBlockedAddrs := app.ModuleAccountAddrs()
-	delete(bankBlockedAddrs, authtypes.NewModuleAddress(
-		providertypes.ConsumerRewardsPool).String())
+	bankBlockedAddrs := BankBlockedAddrs(app)
```

Comment:
The comment above "// Remove ..." should probably be moved to the function `BankBlockedAddrs`.
Also, I would rename the `BankBlockedAddrs` since it does something more than retrieving the blocked addresses, it also deletes one, e.g., something akin to `BankBlockedAddrsWithoutTheConsumerRewardsPool(app)`?

Reply:
changed to ComputeBankBlockedAddrs
I don't like the extremely long name